engine/access/rest/blocks.go

```diff
@@ -0,0 +1,260 @@
+package rest
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/onflow/flow-go/model/flow"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+)
+
+const (
+	ExpandableFieldPayload    = "payload"
+	ExpandableExecutionResult = "execution_result"
+	sealedHeightQueryParam    = "sealed"
+	finalHeightQueryParam     = "final"
+	startHeightQueryParam     = "start_height"
+	endHeightQueryParam       = "end_height"
+	heightQueryParam          = "height"
+)
+
+// getBlocksByID gets blocks by provided ID or list of IDs.
+func getBlocksByIDs(r *requestDecorator, backend access.API, link LinkGenerator) (interface{}, error) {
+
+	ids, err := r.ids()
+	if err != nil {
+		return nil, NewBadRequestError(err)
+	}
+
+	blocks := make([]*generated.Block, len(ids))
+	for i, id := range ids {
+		blkProvider := NewBlockProvider(backend, forID(&id))
+		block, err := getBlock(blkProvider, r, backend, link)
+		if err != nil {
+			return nil, err
+		}
+		blocks[i] = block
+	}
+
+	return blocks, nil
+}
+
+func getBlocksByHeight(r *requestDecorator, backend access.API, link LinkGenerator) (interface{}, error) {
+	heights := r.getQueryParams(heightQueryParam)
+	startHeight := r.getQueryParam(startHeightQueryParam)
+	endHeight := r.getQueryParam(endHeightQueryParam)
+
+	// if both height and one or both of start and end height are provided
+	if len(heights) > 0 && (startHeight != "" || endHeight != "") {
+		err := fmt.Errorf("can only provide either heights or start and end height range")
+		return nil, NewBadRequestError(err)
+	}
+
+	// if neither height nor start and end height are provided
+	if len(heights) == 0 && (startHeight == "" || endHeight == "") {
+		err := fmt.Errorf("must provide either heights or start and end height range")
+		return nil, NewBadRequestError(err)
+	}
+
+	var blocks []*generated.Block
+
+	if len(heights) > 0 {
+		// look up block by heights
+
+		blocks = make([]*generated.Block, len(heights))
+
+		// if the query is /blocks?height=final or /blocks?height=sealed, lookup the last finalized or the last sealed block
+		if heights[0] == finalHeightQueryParam || heights[0] == sealedHeightQueryParam {
+			blkProvider := NewBlockProvider(backend, forFinalized(heights[0]))
+			block, err := getBlock(blkProvider, r, backend, link)
+			if err != nil {
+				return nil, err
+			}
+			blocks[0] = block
+			return blocks, nil
+		}
```

Comment:
that case is not supported.  `final` or `sealed` cannot be combined with another height. Was that something that need to be supported? Is there a use case for it? My assumption was that if client wants final or sealed it will be a request exclusively for that.